### PR TITLE
Move secret key from config to database.

### PR DESCRIPTION
**Pull request by NotSpecial**

### CODE REVIEW

File: amivapi/auth/utils.py [hunk 1]
```diff
@@ -8,6 +8,29 @@
...
+    with app.app_context():  # App context needed for db connection
+        key = 'TOKEN_SECRET'
+        db = app.data.driver.db['config']
+        result = db.find_one({key: {'$exists': True, '$nin': [None, '']}})
+
+        if result is not None:
+            secret = result[key]
+        else:
+            # Create secret and save in db
+            secret = b64encode(urandom(32)).decode('utf_8')
```
Comment (cburchert) [inline amivapi/auth/utils.py]: Not every sequence is valid utf-8, so this might fail somehow.

Comment (cburchert) [reply amivapi/auth/utils.py]: ah I missed the b64encode

Comment (NotSpecial) [reply amivapi/auth/utils.py]: It's first base64 encoded, which can always be utf-8 decoded.

File: amivapi/tests/auth/test_secret.py [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+        super(SecretTest, self).setUp()
+
+        with self.app.app_context():
+            config_value = self.app.config.get(SECRET_KEY)
+            db_item = self.db['config'].find_one({
+                SECRET_KEY: {'$exists': True}
+            })
+
+        self.assertIsNotNone(config_value)
+        self.assertIsNotNone(db_item)
```
Comment (cburchert) [inline amivapi/tests/auth/test_secret.py]: Maybe test for nonempty string as well? Or just assertTrue? This could also pass an empty string otherwise.

Comment (cburchert) [reply amivapi/tests/auth/test_secret.py]: Also why db_item here and db_item[SECRET_KEY] in the next line?

File: amivapi/tests/auth/test_secret.py [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    def setUp(self):
+        """Skip the normal api setup."""
+
+    def test_no_secret(self):
+        """Without init_key, no secret key is in config or db."""
+        # Replace init_secret with a dummy function, which does nothing
+        with patch('amivapi.auth.init_secret'):
+            # Now call setup -- no secret will be initialized
+            super(SecretTest, self).setUp()
+
```
Comment (cburchert) [inline amivapi/tests/auth/test_secret.py]: The with block could end here

File: amivapi/auth/utils.py [hunk 4]
```diff
@@ -8,6 +8,29 @@
 from os import urandom
 
 
+def init_secret(app):
+    """Get secret key from database or create it. Put in config.
+
+    The secret key is stored in the database to ensure consistency.
+    If no secret is in the database yet, create it.
+
+    The database collection holding this key is called `config`
```
Comment (cburchert) [inline amivapi/auth/utils.py]: . is missing at the end of the sentence

File: amivapi/tests/auth/test_secret.py (hunk 3 above)
Comment (NotSpecial) [reply amivapi/tests/auth/test_secret.py]: it does now :)

**Pull request merged**